### ISSUE TITLE
No show/hide callback to nested scene item

### ISSUE THREAD

**Issue by norihiro** [obsproject/obs-studio#5528]
### Operating System Info

Other

### Other OS

Fedora release 34

### OBS Studio Version

Git

### OBS Studio Version (Other)

27.1.3-145-g676ffce4a

### OBS Studio Log URL

https://obsproject.com/logs/nr7cKaErojzDajw1

### Expected Behavior

Source should receive `show` and `hide` callback when the source becomes visible and invisible on preview.

### Current Behavior

Source in a nested scene does not get `show` and `hide` callback when the source becomes visible or invisible on preview by user toggling visibility of the nested scene.

### Steps to Reproduce

1. Set studio mode
2. Create scenes as below
   - `Scene`
   - `Scene 1`
     - `Source`
   - `Scene 2`
     - `Scene 1`
3. Send `Scene` to the program
   Send `Scene 2` to the preview
4. Toggle visibility of `Scene 1` listed as the scene-item of the `Scene 2`

### Anything else we should know?

This bug was originally reported by a user using groups in https://github.com/norihiro/obs-command-source/issues/19.

**Comment by tt2468** [comment]
Does the show callback work when `Scene 2` is sent to the preview?

**Comment by norihiro** [comment]
> Does the show callback work when `Scene 2` is sent to the preview?

Yes, If I switch `Scene` to `Scene 2`, the `show` callback for the `Source` works.
With my several test cases, it looks as if the visibility is always set.
